smslibrary/src/main/java/com/eis/smslibrary/SMSPeer.java

```diff
@@ -4,56 +4,125 @@
+    private static final String INVALID_NUMBER_EXC_MSG = "Invalid telephone number: '%s'. Note that " +
+            "a telephone number pointing outside the currently set default region (%s) should " +
+            "be passed with also the prefix or it might be considered invalid.";
+    private static final String NOT_PARSABLE_EXC_MSG = "Not parsable telephone number: '%s'. Please " +
+            "retry with a different String.";
+    private static final String INVALID_ISO_EXC_MSG = "The given string is not a valid ISO country code.";
 
-    public static final int MAX_TELEPHONE_NUMBER_LENGTH = 15;
-    public static final int MIN_TELEPHONE_NUMBER_LENGTH = 3;
+    // The following regex checks if the given string is one of the following types:
+    // +1#555521555*     [* = any decimal number]
+    // 555521555*        [# = either a space or '-']
+    // +1#555*
+    // 555*
+    // It's used to detect emulator phone numbers
+    private static final String EMU_DETECTION_REGEX = "^(\\+1)?[ -]?(555521)?555\\d$";
 
+    private static String defaultRegion = Locale.getDefault().getCountry();
     private String telephoneNumber;
 
     /**
+     * Set the default country code used when a telephone number without prefix is given.
+     *
+     * @param defaultRegion The default 2 digit ISO country code.
+     * @throws IllegalArgumentException If the given code is not a valid ISO country code.
+     * @author Matteo Carnelos
+     */
+    public static void setDefaultRegion(@NonNull String defaultRegion) {
+        if(!Arrays.asList(Locale.getISOCountries()).contains(defaultRegion))
+            throw new IllegalArgumentException(INVALID_ISO_EXC_MSG);
+        SMSPeer.defaultRegion = defaultRegion;
+    }
+
+    /**
+     * Create a new SMSPeer object. If the given number does not explicitly contains the prefix
+     * will be added the default one.
+     * Note: It is not always possible to obtain the correct default country code, especially if the
+     * phone is moving between countries. To set a custom one use the method {@link #setDefaultRegion(String)}.
+     *
      * @param telephoneNumber Address for the peer.
-     * @throws InvalidTelephoneNumberException If telephoneNumber check is not {@link TelephoneNumberState#TELEPHONE_NUMBER_VALID}.
+     * @throws InvalidTelephoneNumberException If telephoneNumber is not valid.
+     * @throws IllegalArgumentException If the given telephoneNumber is not parsable.
+     * @author Matteo Carnelos
      */
-    public SMSPeer(String telephoneNumber) throws InvalidTelephoneNumberException {
-        TelephoneNumberState telephoneNumberState = SMSPeer.checkPhoneNumber(telephoneNumber);
-        if(telephoneNumberState != TelephoneNumberState.TELEPHONE_NUMBER_VALID)
-            throw new InvalidTelephoneNumberException(telephoneNumberState);
+    public SMSPeer(@NonNull String telephoneNumber) throws InvalidTelephoneNumberException {
```

Comment:
Also, doesn't this method throw `IllegalArgumentException` because of `parseNumber(...)`?

Reply:
Oh wait, you got it right in the specifications, add it to the `throws` list of exceptions too, it's cool.

Reply:
You know what? I think that is better if we remove the `throws` clause here and also in `parseNumber(...)`. It's reccomended to declare in the method signature only checked exceptions, unchecked ones like these don't need to. They can be declared only in the throws tag in my opinion.